corehq/apps/domain/management/commands/reset_user_passwords.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import print_function
+from __future__ import absolute_import
+from bulk_update.helper import bulk_update as bulk_update_helper
+from django.contrib.auth.models import User
+from django.core.management.base import BaseCommand
+
+
+from corehq.apps.users.models import CommCareUser
+from corehq.util.view_utils import absolute_reverse
+from dimagi.utils.chunked import chunked
+from six.moves import input
+
+
+class Command(BaseCommand):
+    help = "Reset all mobile user passwords to a given password"
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            'domain_name',
+        )
+        parser.add_argument(
+            'password',
+        )
+
+    def handle(self, domain_name, password, **options):
+        url = absolute_reverse('domain_homepage', args=[domain_name])
+        confirm = input(
+            """
+            ## Caution!! ##
+            ###############
+            You are resetting passwords of all users in {domain} at {url}
+            Are you sure want to reset, type your domain name to confirm?
+            """.format(domain=domain_name, url=url)
+        )
+        if not confirm == domain_name:
+            print("Aborted")
+            return
+
+        all_users = CommCareUser.by_domain(domain_name)
+        django_users = User.objects.filter(
+            username__endswith="@{domain}.commcarehq.org".format(domain=domain_name)
+        )
+        assert len(all_users) == django_users.count(), "Something is wrong, django/couch users don't match"
+
+        num_users = len(all_users)
+        for i, users in enumerate(chunked(all_users, 1000)):
+            for user in users:
+                user.set_password(password)
+            CommCareUser.bulk_save(users)
```

Comment:
does this work with all the code we have to update both couch and django users?

Reply:
Yea, there isn't extra things that needs to happen on password change. There are couch/django syncs, but that's taken care of by doing it separately. There are post-save signals that update ES, but it's not needed for password change.